processor/processor.go

```diff
@@ -0,0 +1,242 @@
+package processor
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/algorand/go-algorand-sdk/client/v2/algod"
+	"github.com/algorand/go-algorand/agreement"
+	"github.com/algorand/go-algorand/config"
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/bookkeeping"
+	"github.com/algorand/go-algorand/data/transactions"
+	"github.com/algorand/go-algorand/ledger"
+	"github.com/algorand/go-algorand/ledger/ledgercore"
+	"github.com/algorand/go-algorand/logging"
+	"github.com/algorand/go-algorand/protocol"
+	"github.com/algorand/go-algorand/rpcs"
+	"github.com/algorand/indexer/accounting"
+)
+
+type Processor interface {
+	Process(cert *rpcs.EncodedBlockCert) error
+	SetHandler(handler func(block *ledgercore.ValidatedBlock) error)
+	GetLastProcessedRound() uint64
+	Start(path string, client *algod.Client)
+}
+
+type ProcessorImpl struct {
+	handler            func(block *ledgercore.ValidatedBlock) error
+	lastProcessedRound uint64
+	ledger             *ledger.Ledger
+}
+
+// Start block processor. Create ledger if it wasn't initialized before
+func (processor *ProcessorImpl) Start(path string, client *algod.Client) {
+	//open ledger
+	genesis, err := getGenesis(client)
+	if err != nil {
+		fmt.Printf("error getting genesis err: %v", err)
+		os.Exit(1)
+	}
+	genesisBlock, err := getGenesisBlock(client)
+	if err != nil {
+		fmt.Printf("error getting genesis block err: %v", err)
+		os.Exit(1)
+	}
+	openLedger(path, &genesis, &genesisBlock)
+}
+
+// Process a raw algod block
+func (processor *ProcessorImpl) Process(cert *rpcs.EncodedBlockCert) error {
+	if processor.ledger == nil {
+		panic(fmt.Errorf("Process() err: local ledger not initialized"))
+	}
+	if uint64(cert.Block.Round()) != processor.lastProcessedRound+1 {
+		return fmt.Errorf("Process() err: block has invalid round")
+	}
+	if cert.Block.Round() == basics.Round(0) {
+		// Block 0 is special, we cannot run the evaluator on it
+		err := processor.ledger.AddBlock(cert.Block, agreement.Certificate{})
+		if err != nil {
+			return fmt.Errorf("error adding round  %v to local ledger", err.Error())
+		}
+		processor.lastProcessedRound = uint64(cert.Block.Round())
+	} else {
+		proto, ok := config.Consensus[cert.Block.BlockHeader.CurrentProtocol]
+		if !ok {
+			return fmt.Errorf(
+				"Process() cannot find proto version %s", cert.Block.BlockHeader.CurrentProtocol)
+		}
+		proto.EnableAssetCloseAmount = true
+
+		ledgerForEval, err := MakeLedgerForEvaluator(processor.ledger)
+		if err != nil {
+			return fmt.Errorf("Process() err: %w", err)
+		}
+		defer ledgerForEval.Close()
+		resources, _ := prepareEvalResources(&ledgerForEval, &cert.Block)
+		if err != nil {
+			panic(fmt.Errorf("Process() eval err: %w", err))
+		}
+
+		delta, _, err :=
+			ledger.EvalForIndexer(ledgerForEval, &cert.Block, proto, resources)
```

Comment:
Did you look into using StartEvaluator instead of EvalForIndexer? It's interesting that by implementing the `LedgerForEval` interface we could theoretically swap between the local ledger and postgres, but I'm not sure if we would ever want to do that.